pkgs/build-support/fetchnextcloudapp/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, gnutar, findutils, fetchurl, ... }:
+{ name
+, url
+, version
+, sha256
+, patches ? [ ]
+}:
+stdenv.mkDerivation {
+  name = "nc-app-${name}";
+  inherit version patches;
+
+  src = fetchurl {
+    url = url;
+    sha256 = sha256;
+  };
+
+  nativeBuildInputs = [
+    gnutar
+    findutils
+  ];
+
+  unpackPhase = ''
+    tar -xzpf $src
+    find -type f -executable -exec chmod o-x,g-x-w,a-x-w '{}' \;
```

Comment:
I think the easiest solution is a `chmod -R go-w` here to leave it out below in `installPhase` entirely :)

Reply:
Done in 98f95f9